pkg/common/cns-lib/volume/manager.go

```
@@ -2795,13 +2795,18 @@ func (m *defaultManager) createSnapshotWithImprovedIdempotencyCheck(ctx context.
 // This function ensures that no orphaned snapshots are left behind on the vSphere backend
 // in case of failures during the snapshot creation process
 func (m *defaultManager) createSnapshotWithTransaction(ctx context.Context, volumeID string,
-	snapshotID string, extraParams interface{}) (*CnsSnapshotInfo, string, error) {
+	snapshotName string, extraParams interface{}) (*CnsSnapshotInfo, string, error) {
 	log := logger.GetLogger(ctx)
 	var (
 		// Reference to the CreateSnapshot task on CNS.
 		createSnapshotsTask *object.Task
 		// Name of the CnsVolumeOperationRequest instance.
-		instanceName = snapshotID + "-" + volumeID
+		instanceName = snapshotName + "-" + volumeID
+		// By default, external-snapshotter sets the snapshot name prefix to "snapshot-".
+		// This logic will break if the prefix configuration is changed.
+		// In Supervisor deployments, we assume this configuration remains unchanged by admin/DevOps.
+		// In Vanilla deployments, we publish the deployment manifest with the default configuration to ensure consistency.
+		snapshotID = strings.TrimPrefix(snapshotName, "snapshot-")
```

Can you fail the operation if "snapshot-" is not found?